src/libsyntax/parse/parser.rs

```diff
@@ -4070,6 +4088,9 @@ impl<'a> Parser<'a> {
                     TraitBoundModifier::None
                 };
                 bounds.push(TraitTyParamBound(poly_trait, modifier));
+                if has_parens {
+                    self.expect(&token::CloseDelim(token::Paren))?;
+                }
             } else {
                 break
```

Comment:
So, we consumed an open-paren at the start of the loop, don't we want to consume the close paren here somewhere?